### PR TITLE
Left over volume snapshot class

### PR DESCRIPTION
**Pull request by bathina2**
Using v1alpha1 apis for delete calls was making it fail.

### CODE REVIEW

File: pkg/csi/snapshot_restore.go [hunk 1]
```diff
@@ -488,24 +507,31 @@ func (c *snapshotCreate) CreateSnapshot(ctx context.Context, snapshotter kansnap
...
-	defer func() {
-		_ = c.dynCli.Resource(v1alpha1.VolSnapClassGVR).Delete(ctx, targetSnapClassName, metav1.DeleteOptions{})
-	}()
 	err := snapshotter.CloneVolumeSnapshotClass(args.VolumeSnapshotClass, targetSnapClassName, kansnapshot.DeletionPolicyRetain, nil)
 	if err != nil {
 		return errors.Wrapf(err, "Failed to create a VolumeSnapshotClass to use to restore the snapshot")
 	}
+	defer func() {
+		VolSnapClassGVR := schema.GroupVersionResource{Group: SnapGroupName, Version: SnapshotGroupVersion.Version, Resource: VolumeSnapshotClassResourcePlural}
+		err := c.dynCli.Resource(VolSnapClassGVR).Delete(ctx, targetSnapClassName, metav1.DeleteOptions{})
```
Comment (bathina2) [inline pkg/csi/snapshot_restore.go:530]: Passing the built GVR instead of `v1alpha1.VolSnapClassGVR`

File: pkg/csi/snapshot_restore.go [hunk 2]
```diff
@@ -135,12 +137,13 @@ type SnapshotRestoreStepper interface {
...
-	dataValidatorOps  DataValidator
-	snapshotCreateOps SnapshotCreator
-	cleanerOps        Cleaner
+	validateOps          ArgumentValidator
+	versionFetchOps      ApiVersionFetcher
+	createAppOps         ApplicationCreator
+	dataValidatorOps     DataValidator
+	snapshotCreateOps    SnapshotCreator
+	cleanerOps           Cleaner
+	SnapshotGroupVersion *metav1.GroupVersionForDiscovery
```
Comment (redgoat650) [inline pkg/csi/snapshot_restore.go:146]: Does it need to be exported?
```suggestion
	snapshotGroupVersion *metav1.GroupVersionForDiscovery
```

File: pkg/csi/snapshot_restore.go [hunk 3]
```diff
@@ -452,7 +471,7 @@ func (c *applicationCreate) WaitForPodReady(ctx context.Context, namespace strin
 type SnapshotCreator interface {
 	NewSnapshotter() (kansnapshot.Snapshotter, error)
 	CreateSnapshot(ctx context.Context, snapshotter kansnapshot.Snapshotter, args *types.CreateSnapshotArgs) (*v1alpha1.VolumeSnapshot, error)
-	CreateFromSourceCheck(ctx context.Context, snapshotter kansnapshot.Snapshotter, args *types.CreateFromSourceCheckArgs) error
+	CreateFromSourceCheck(ctx context.Context, snapshotter kansnapshot.Snapshotter, args *types.CreateFromSourceCheckArgs, SnapshotGroupVersion *metav1.GroupVersionForDiscovery) error
```
Comment (redgoat650) [inline pkg/csi/snapshot_restore.go:474]:
```suggestion
	CreateFromSourceCheck(ctx context.Context, snapshotter kansnapshot.Snapshotter, args *types.CreateFromSourceCheckArgs, snapshotGroupVersion *metav1.GroupVersionForDiscovery) error
```

File: pkg/csi/snapshot_restore.go [hunk 4]
```diff
@@ -532,7 +558,7 @@ func (c *snapshotCreate) CreateFromSourceCheck(ctx context.Context, snapshotter
 type Cleaner interface {
 	DeletePVC(ctx context.Context, pvcName string, namespace string) error
 	DeletePod(ctx context.Context, podName string, namespace string) error
-	DeleteSnapshot(ctx context.Context, snapshotName string, namespace string) error
+	DeleteSnapshot(ctx context.Context, snapshotName string, namespace string, SnapshotGroupVersion *metav1.GroupVersionForDiscovery) error
```
Comment (redgoat650) [inline pkg/csi/snapshot_restore.go:561]:
```suggestion
	DeleteSnapshot(ctx context.Context, snapshotName string, namespace string, snapshotGroupVersion *metav1.GroupVersionForDiscovery) error
```

**Review by redgoat650 — COMMENTED** [review]
Some convention suggestions. Seems alright to me overall

**Pull request merged**